### DISCUSSION TITLE
Enabling scrolling in a DataTable

### DISCUSSION THREAD

**Discussion by Sim0nRoden** [flet-dev/flet#1152] [Q&A] [answered]
### Question

Hi,
Im searching for a function to enable scrolling in a DataTable since my rows in it are going out of the page. I already did try the things listed in [this post](https://stackoverflow.com/questions/74896789/python-flet-scroll-mode-for-datatable) but it didnt work for me. Is there a function that im missing?

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ndonkoHenri** [comment, marked as answer]
There is no official way to scroll at the moment, but a hacky/nonofficial way is to put the datatable control in a Row with scroll enabled.

Let me know if it worked as expected.

**Reply by vindevoy** [reply]
I think a better way, for vertical scrolling, is the list view

**Comment by Sim0nRoden** [comment]
Hi,
thank you for your answer! I want vertical scrolling so it would need to enable scroll in a column.
My view currently looks like this:
![image](https://user-images.githubusercontent.com/76072848/223683498-a3e2c7dc-be7f-4145-bbaa-9d1f42897487.png)
I want the scroll only for the table and not the whole page since I intend to add another table right next to this one.
I tried adding a column to the Row of the table like this:
![image](https://user-images.githubusercontent.com/76072848/223698982-bc2d1463-4e8a-4c89-b004-c73972af263e.png)
but then my table is not shown on the view. I'm not a pro with flet just trying things out so I may have some useless Controls in my view or smth.
Heres my table if you need it:
![image](https://user-images.githubusercontent.com/76072848/223699368-4cb5eafc-ef9b-4764-b544-c40f3e567dee.png)
Thanks for the help

**Reply by ndonkoHenri** [reply]
It should be because you expanded the `DataTable`.

Try expanding(`expand=True`) just the `Row`, the `Column` or both.

Doing this might maybe cause the need for other controls to have it too. 🤔

If my suggestion doesn't work, paste some little sample code here so I can help you better.

**Reply by Sim0nRoden** [reply]
I managed to get the table to show up by removing the expand from it as you said, adding the expand to the other controls (Row/Column) will result in the table not showing up again. Adding the scroll property to the Column/Row didnt add a scrollbar for me (maybe im doing something wrong, idk). Sharing sample code will be quite difficult because the table (and other objects aswell) call different functions in other files, putting them all together would be to much work... but here is the code from the above screenshots in written form. These are just excerpts of some parts of my app so it wont run 

        self.brokerDropdown = ft.Dropdown(
            label="Broker",
            width=200,
            options=[
                ft.dropdown.Option("Historic-Papertrade"),
                ft.dropdown.Option("Broker2"),
                ft.dropdown.Option("Broker3"),
            ],
            value="Historic-Papertrade"
        )

        self.addStockButton = ft.IconButton(
            icon=ft.icons.ADD,
            on_click=self.addStockDialog,
            icon_color=ft.colors.BLUE_GREY_400,
            selected=False,
        )

        self.table = ft.DataTable(
            # expand=True,
            border=ft.border.all(2, "grey"),
            border_radius=10,
            vertical_lines=ft.border.BorderSide(3, "red"),
            horizontal_lines=ft.border.BorderSide(3, "blue"),
            show_checkbox_column=True,
            sort_column_index=0,
            sort_ascending=True,
            columns=[
                ft.DataColumn(
                    ft.Icon(ft.icons.STAR),
                    tooltip="Favorites the entry)",
                    on_sort=self.sort,
                ),
                ft.DataColumn(
                    ft.Text("Symbol"),
                    tooltip="The symbol of the desired stock (i.e. Apple=AAPL, Microsoft=MSFT)",
                    on_sort=self.sort,
                ),
                ft.DataColumn(
                    ft.Text("Trade limit"),
                    tooltip="The amount of money that the stock can trade with",
                    on_sort=self.sort,
                ),
                ft.DataColumn(
                    ft.Icon(ft.icons.DELETE),
                    tooltip="Deletes the entry",
                ),
            ],
            rows=self.row
        )

        self.view = ft.Column([
            ft.Container(
                ft.Column([
                    ft.Row([
                        self.brokerDropdown
                    ]),
                    ft.Row([
                        self.stockTable.addStockButton
                    ]),
                    ft.Row([
                        ft.Column([
                            self.stockTable.table,
                        ]),
                    ]),
                ]),
                padding=ft.padding.only(top=40, bottom=40, right=40)
            )
        ])

**Reply by Sim0nRoden** [reply]
Update: I removed the top Column of the view and the table still showed up (I guess it was just useless). After that I tried to add the scrolling and expand to it and I managed to get scrolling to work. I hope that I can somehow get the table header to always show and the scrolling is just for the rows, but thats optional.  Now I have the problem that the table is not expanding (adding expand to the column of the table will move the scrollbar to the far left (it expands) but the table stays in the normal size (does not expand). Here is waht the view currently looks like: 

        self.view = ft.Container(
            ft.Column([
                ft.Row([
                    self.brokerDropdown
                ]),
                ft.Row([
                    self.stockTable.addStockButton
                ]),
                ft.Row([
                    ft.Column([
                        self.stockTable.table,
                    ], scroll="adaptive"),
                ], expand=True),
            ]),
            padding=ft.padding.only(top=40, bottom=40, right=40)
        )

**Reply by batagope** [reply]
You can use ListView to enable scroll for Datatable

```
datatable = ft.DataTable(...)
lv = ft.ListView(expand=1, auto_scroll=True, padding=10)
lv.controls.append(datatable)

page.add(lv)

```

**Reply by Ben2022-cell** [reply]
If we add  ListView to the Row(), that DataTable didn't scroll, how can be method to solution?

**Reply by sukarnar** [reply]
To enable scrolling to DataTable use Listview like below
ft.ListView(expand=1, auto_scroll=True, padding=10, controls=[ft.DataTable()])

**Reply by Ben2022-cell** [reply]
thank your answer.

**Comment by vindevoy** [comment]
I have tried the solution with the list view and that works, but only for vertical scrolling.  My datatable has too many columns to show, but I don't get a horizontal scrollbar.
